File: MedArrival-Backend/src/main/java/com/cnesten/medarrivalbackend/DTO/ReceiptDTO.java
package com.cnesten.medarrivalbackend.DTO;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Data
public class ReceiptDTO {
    private Long id;
    private String receiptNumber;
    private LocalDateTime receiptDate;
    private Float totalAmount;
    private String iceNumber;
    private String referenceNumber;
    private String deliveryNoteNumbers;
    private Float tvaPercentage;
    private Float totalHT;
    private Float totalTTC;
    private String paymentTerms;
    private String bankAccount;
    private String bankDetails;
    private String issuingDepartment;
    private String deliveryRef;
    private Boolean deliveryReceived;
    private ClientDTO client;
    private Set<ReceiptItemDTO> receiptItems = new HashSet<>();
    private Set<ReceiptAttachmentDTO> attachments = new HashSet<>();
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}

File: MedArrival-Front/src/models/ReceiptDTO.ts
import { BaseDTO } from "./BaseDTO";
import { ClientDTO } from "./ClientDTO";
import { ProductDTO } from "./ProductDTO";

// Receipt Item DTO
export interface ReceiptItemDTO extends BaseDTO {
    product: ProductDTO;
    quantity: number;
    unitPrice: number;
    subtotal: number;
    lotNumber: string | null;
    calibrationDate: Date | null;
    expirationDate: Date | null;
    articleCode: string | null;
    description: string | null;
    unit: string | null;
}

// Receipt DTO
export interface ReceiptDTO extends BaseDTO {
    receiptNumber: string;
    receiptDate: Date;
    totalAmount: number;
    iceNumber: string | null;
    referenceNumber: string | null;
    deliveryNoteNumbers: string | null;
    tvaPercentage: number | null;
    totalHT: number | null;
    totalTTC: number | null;
    paymentTerms: string | null;
    bankAccount: string | null;
    bankDetails: string | null;
    issuingDepartment: string | null;
    deliveryRef: string | null;
    deliveryReceived: boolean | null;
    client: ClientDTO;
    receiptItems: ReceiptItemDTO[];
    attachments: ReceiptAttachmentDTO[];
}

// Receipt Attachment DTO
export interface ReceiptAttachmentDTO extends BaseDTO {
    fileName: string;
    originalName: string;
    fileType: string;
    fileSize: number; 
}